src/SpectatorView.Unity/Assets/SpectatorView/Scripts/Compositor/StationaryCameraCalibrationDataProvider.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.SpectatorView
+{
+    public class StationaryCameraCalibrationDataProvider : MonoBehaviour
+    {
+        private INetworkManager networkManager;
+
+        private void Awake()
+        {
+            networkManager = GetComponent<INetworkManager>();
+            if (networkManager == null)
+            {
+                throw new MissingComponentException("Missing network manager component");
+            }
+
+            networkManager.Connected += NetworkManagerConnected;
+            if (networkManager.IsConnected)
+            {
+                SendCalibrationData();
+            }
+        }
+
+        private void OnDestroy()
+        {
+            networkManager.Connected -= NetworkManagerConnected;
+        }
+
+        private void NetworkManagerConnected(INetworkConnection obj)
+        {
+            SendCalibrationData();
+        }
+
+#if UNITY_EDITOR && UNITY_WSA
+        private void SendCalibrationData()
```

Comment:
does the azure kinect sdk work with UNITY_WSA? do we have any scenarios where we would build a UNITY_WSA stationary camera calibration data provider?

Reply:
Really this should only happen in editor now, the UNITY_WSA check is probably not necessary. I'll remove it and also log a "Platform not supported" error message in the other SendCalibrationData block.

---
In reply to: [384190170](https://github.com/microsoft/MixedReality-SpectatorView/pull/327#discussion_r384190170) [](ancestors = 384190170)